torch/autograd/profiler.py

```diff
@@ -116,55 +116,56 @@ def export_chrome_trace(self, path):
             chrome_events = []
             next_id = 0
+            # Use file IO over using json.dump since JSON dumping is very slow and
+            # this technique is proven to give a 4x speedup.
+            f.write("[")
             for evt in self:
-                chrome_events.append(dict(
-                    name=evt.name,
-                    ph='X',
-                    ts=evt.cpu_interval.start,
-                    dur=evt.cpu_interval.elapsed_us(),
-                    tid=evt.thread,
-                    pid='CPU functions',
-                    args={},
-                ))
+                f.write('{"name": "%s", '
+                        '"ph": "X", '
+                        '"ts": %s, '
+                        '"dur": %s, '
+                        '"tid": %s, '
+                        '"pid": "CPU functions", '
+                        '"args": {}}, ' % (evt.name, evt.cpu_interval.start,
+                                           evt.cpu_interval.elapsed_us(), evt.thread))
                 for k in evt.kernels:
                     # 's' and 'f' draw Flow arrows from
                     # the CPU launch to the GPU kernel
-                    chrome_events.append(dict(
-                        name=evt.name,
-                        ph='s',
-                        ts=evt.cpu_interval.start,
-                        tid=evt.thread,
-                        pid='CPU functions',
-                        id=next_id,
-                        cat='cpu_to_cuda',
-                        args={},
-                    ))
-                    chrome_events.append(dict(
-                        name=k.name,
-                        ph='f',
-                        ts=k.interval.start,
-                        tid=k.device,
-                        pid='CUDA functions',
-                        id=next_id,
-                        cat='cpu_to_cuda',
-                        args={},
-                    ))
-                    chrome_events.append(dict(
-                        name=k.name,
-                        ph='X',
-                        ts=k.interval.start,
-                        dur=k.interval.elapsed_us(),
-                        tid=k.device,
-                        pid='CUDA functions',
-                        args={},
-                    ))
+                    f.write('{"name": "%s", '
+                            '"ph": "s", '
+                            '"ts": %s, '
+                            '"tid": %s, '
+                            '"pid": "CPU functions", '
+                            '"id": %s, '
+                            '"cat": "cpu_to_cuda", '
+                            '"args": {}}, ' % (evt.name, evt.cpu_interval.start,
+                                               evt.thread, next_id))
+                    f.write('{"name": "%s", '
+                            '"ph": "f", '
+                            '"ts": %s, '
+                            'tid": %s, '
```

Comment:
I'm pretty sure this is invalid JSON and there needs to be a `"` before tid here. As a result, I don't think export_chrome_trace with the profiler using `use_cuda=True` has been working since this PR. Fixing in https://github.com/pytorch/pytorch/pull/33987/files

Reply:
Thanks @rohan-varma, you are correct this is a bug. Note that there was an existing test for `export_chrome_trace` (`test_profiler_aggregation_lstm` in `test_autograd.py`), but in review we missed that that doesn't cover CUDA.

Reply:
I think that test is insufficient in that it also wouldn't catch this since it doesn't check the output of `export_chrome_trace` to ensure the JSON is valid?